Improve latency for topics

Always use the TCP_NODELAY socket options when subscribing to topics, either in
the C client or for input ports in a server. It seems to improve latency a lot
for small messages (~600bytes).

diff --git client/c/client.cc client/c/client.cc
index 629c880..99f6056 100644
--- client/c/client.cc
+++ client/c/client.cc
@@ -288,7 +288,9 @@ genom_<"$comp">_client_<"[$p name]">_port(
     p->data.reset();
     p->s = h->node->subscribe< genom::port::<"[$p name]"> >(
       pname, 1,
-      boost::bind(&<"$comp">_<"[$p name]">_port_cb, h->topics_timeout, p, _1));
+      boost::bind(&<"$comp">_<"[$p name]">_port_cb, h->topics_timeout, p, _1),
+      ros::VoidConstPtr(),
+      ros::TransportHints().tcpNoDelay());
     if (!p->s) {
       return genom_no_such_outport(h);
     }
diff --git server/port.h server/port.h
index c666267..70af073 100644
--- server/port.h
+++ server/port.h
@@ -107,7 +107,8 @@ struct genom_port_<"[$p name]">_base {
     if (i == topics.end()) return 1;

     sub = node->subscribe(
-      absname, 1, &genom_port_<"[$p name]">_base::datacb, this);
+      absname, 1, &genom_port_<"[$p name]">_base::datacb, this,
+      ros::TransportHints().tcpNoDelay());
     return !sub;
   }
